Source/WebCore/css/StyleProperties.cpp

```diff
@@ -705,6 +707,68 @@ String StyleProperties::fontVariantValue() const
     return result.toString();
 }
 
+String StyleProperties::fontSynthesisValue() const
+{
+    StringBuilder result;
+
+    auto getExplicitLonghandValue = [&](CSSPropertyID propertyID) -> CSSValue* {
+        auto foundPropertyIndex = findPropertyIndex(propertyID);
+        if (foundPropertyIndex == -1)
+            return nullptr;
+
+        auto property = propertyAt(foundPropertyIndex);
+        if (property.isImplicit())
+            return nullptr;
+
+        return property.value();
+    };
+
+    auto appendWithPrefixIfNeeded = [&](ASCIILiteral word) {
+        if (!result.isEmpty())
+            result.append(' ');
+        result.append(word);
+    };
+    
+    // font-synthesis: none | [ weight || style || small-caps ]
+    auto weightValue = getExplicitLonghandValue(CSSPropertyFontSynthesisWeight);
+    auto styleValue = getExplicitLonghandValue(CSSPropertyFontSynthesisStyle);
+    auto capsValue = getExplicitLonghandValue(CSSPropertyFontSynthesisSmallCaps);
+
+    auto valueID = [&](CSSValue *value) {
+        if (!value)
+            return CSSValueInvalid;
+        return downcast<CSSPrimitiveValue>(value)->valueID();
+    };
+
+    auto weightValueID = valueID(weightValue);
+    auto styleValueID = valueID(styleValue);
+    auto capsValueID = valueID(capsValue);
+
+    if (weightValue && styleValue && capsValue) {
+        if (*weightValue == *styleValue && *weightValue == *capsValue) {
+            // font-synthesis: value can be none or CSS wide value keyword
+            if (isCSSWideValueKeyword(weightValue->cssText()) || weightValueID == CSSValueNone)
+                return weightValue->cssText();
+        }
+    }
```

Comment:
This would be shorter if you just compare value IDs:

```suggestion
    if (weightValueID == styleValueID && weightValueID == capsValueID) {
        // font-synthesis: value can be none or CSS wide value keyword
        if (isCSSWideValueKeyword(weightValue->cssText()) || weightValueID == CSSValueNone)
            return weightValue->cssText();
    }
```